### PR TITLE
Brod 9 acking

### PR DESCRIPTION
**Pull request by jeffgrunewald**
Offloads acknowledgement for group consumers to a separate process to bypass the group manager being a bottleneck for acking messages during an assignments revoked event.

Updates dependencies and documentation.

### CODE REVIEW

File: lib/elsa/group/acknowledger.ex [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      group_coordinator_pid: nil
+    }
+
+    {:ok, state, {:continue, :get_coordinator}}
+  end
+
+  @impl GenServer
+  def handle_continue(:get_coordinator, state) do
+    retriever = fn -> Elsa.Registry.whereis_name({registry(state.connection), :brod_group_coordinator}) end
+    group_coordinator_pid = retrieve_coordinator(retriever, 5)
```
Comment (bbalser) [inline lib/elsa/group/acknowledger.ex]: What if we move the brod_group_coordinator under the Elsa.Group.Supervisor (wrapped by Elsa.Wrapper)?  That should remove the need for the retrieve_coordinator loop and simplify the death handling login in Elsa.Group.Manager as well.

File: lib/elsa/group/manager.ex [hunk 2]
```diff
@@ -118,28 +118,6 @@ defmodule Elsa.Group.Manager do
     GenServer.call(pid, :revoke_assignments, 30_000)
   end
 
-  @doc """
```
Comment (bbalser) [inline lib/elsa/group/manager.ex]: Should we add a defdelegate to Acknowledger to keep backwards compatibility?

File: lib/elsa/group/manager.ex [hunk 3]
```diff
@@ -186,6 +164,9 @@ defmodule Elsa.Group.Manager do
         {:stop, reason, {:error, reason}, state}
 
       :ok ->
+        Elsa.Registry.whereis_name({registry(state.connection), Elsa.Group.Acknowledger})
```
Comment (bbalser) [inline lib/elsa/group/manager.ex]: via tuple?

File: lib/elsa/group/manager/worker_manager.ex [hunk 4]
```diff
@@ -59,8 +50,11 @@ defmodule Elsa.Group.Manager.WorkerManager do
   def restart_worker(workers, ref, %Elsa.Group.Manager.State{} = state) do
     worker = get_by_ref(workers, ref)
 
-    assignment =
-      brod_received_assignment(topic: worker.topic, partition: worker.partition, begin_offset: worker.latest_offset)
+    latest_offset =
+      Elsa.Registry.whereis_name({registry(state.connection), Elsa.Group.Acknowledger})
```
Comment (bbalser) [inline lib/elsa/group/manager/worker_manager.ex]: via tuple?

File: lib/elsa/group/manager.ex (hunk 2 above)
Comment (jeffgrunewald) [reply lib/elsa/group/manager.ex]: Didn’t realize that was there. Sure thing.

File: lib/elsa/group/acknowledger.ex (hunk 1 above)
Comment (jeffgrunewald) [reply lib/elsa/group/acknowledger.ex]: I’ll give it a try and see how it goes.

File: lib/elsa/group/manager.ex (hunk 2 above)
Comment (jeffgrunewald) [reply lib/elsa/group/manager.ex]: Never mind; I just hadn't seen delegates outside the context of a top-level module passing down to deeper modules. Understood and added.

File: lib/elsa/group/acknowledger.ex (hunk 1 above)
Comment (jeffgrunewald) [reply lib/elsa/group/acknowledger.ex]: Turns out it works.

**Pull request merged**